File: Freedom/微信/基本类库/Common/WXSearchController.m
//  TLSearchController.m
//  Freedom
// Created by Super
#import "WXSearchController.h"
@implementation WXSearchController
- (id)initWithSearchResultsController:(UIViewController *)searchResultsController{
    if (self = [super initWithSearchResultsController:searchResultsController]) {
        [self.searchBar setFrame:CGRectMake(0, 0, APPW, TopHeight)];
        [self.searchBar setBackgroundImage:[FreedomTools imageWithColor:[UIColor lightGrayColor]]];
        [self.searchBar setBarTintColor:[UIColor lightGrayColor]];
        [self.searchBar setTintColor:[UIColor greenColor]];
        UITextField *tf = [[[self.searchBar.subviews firstObject] subviews] lastObject];
        [tf.layer setMasksToBounds:YES];
        [tf.layer setBorderWidth:1];
        [tf.layer setBorderColor:[UIColor grayColor].CGColor];
        [tf.layer setCornerRadius:5.0f];
        
        for (UIView *view in self.searchBar.subviews[0].subviews) {
            if ([view isKindOfClass:NSClassFromString(@"UISearchBarBackground")]) {
                UIView *bg = [[UIView alloc] init];
                [bg setBackgroundColor:[UIColor lightGrayColor]];
                [view addSubview:bg];
                [bg mas_makeConstraints:^(MASConstraintMaker *make) {
                    make.edges.mas_equalTo(0);
                }];
                
                UIView *line = [[UIView alloc] init];
                [line setBackgroundColor:[UIColor grayColor]];
                [view addSubview:line];
                [line mas_makeConstraints:^(MASConstraintMaker *make) {
                    make.left.and.right.and.bottom.mas_equalTo(0);
                    make.height.mas_equalTo(1);
                }];
                break;
            }
        }
    }
    return self;
}
- (void)setShowVoiceButton:(BOOL)showVoiceButton{
    _showVoiceButton = showVoiceButton;
    if (showVoiceButton) {
        [self.searchBar setShowsBookmarkButton:YES];
        [self.searchBar setImage:[UIImage imageNamed:@"searchBar_voice"] forSearchBarIcon:UISearchBarIconBookmark state:UIControlStateNormal];
        [self.searchBar setImage:[UIImage imageNamed:@"searchBar_voice_HL"] forSearchBarIcon:UISearchBarIconBookmark state:UIControlStateHighlighted];
    }else{
        [self.searchBar setShowsBookmarkButton:NO];
    }
}
@end

File: Freedom/微信/基本类库/Common/WXSearchController.swift
//
//  WXSearchController.swift
//  Freedom

import Foundation
import SnapKit
//import XExtension
class WXSearchController: UISearchController {
    var showVoiceButton = false {
        didSet {
            if showVoiceButton {
                searchBar.showsBookmarkButton = true
                searchBar.setImage(WXImage.barVoice.image, for: .bookmark, state: .normal)
                searchBar.setImage(WXImage.barVoice.image, for: .bookmark, state: .highlighted)
            } else {
                searchBar.showsBookmarkButton = false
            }
        }
    }
    override init(searchResultsController: UIViewController?) {
        super.init(searchResultsController: searchResultsController)
        searchBar.frame = CGRect(x: 0, y: 0, width: APPW, height: 44)
        searchBar.backgroundImage = UIImage.imageWithColor(.clear)
        searchBar.barTintColor = UIColor.light
        searchBar.tintColor = UIColor.greenx
        let tf = searchBar.subviews.first?.subviews.last as! UITextField
        tf.layer.masksToBounds = true
        tf.layer.borderWidth = 1
        tf.layer.borderColor = UIColor.grayx.cgColor
        tf.layer.cornerRadius = 5.0
        for view: UIView in searchBar.subviews[0].subviews {
            if (NSStringFromClass(object_getClass(view) ?? UIView.self) == "UISearchBarBackground") {
                let bg = UIView()
                bg.backgroundColor = UIColor.light
                view.addSubview(bg)
                bg.snp.makeConstraints { (make) in
                    make.edges.equalToSuperview()
                }
                let line = UIView()
                line.backgroundColor = UIColor.grayx
                view.addSubview(line)
                line.snp.makeConstraints { (make) in
                    make.left.right.bottom.equalToSuperview()
                    make.height.equalTo(1)
                }
                break
            }
        }
    }
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
